src/pystage/core/assets.py

```diff
@@ -53,7 +55,30 @@ def next_costume(self):
             self.current_costume = 0
         self.update_sprite_image()
 
+    def previous_costume(self):
+        if self.current_costume == -1:
+            return
+        self.current_costume -= 1
+        if self.current_costume == -1:
+            self.current_costume = len(self.costumes) - 1
+        self.update_sprite_image()
+
+    def random_costume(self):
+        if self.current_costume == -1:
+            return
+        self.current_costume = random.randint(0, len(self.costumes)-1)
+        self.update_sprite_image()
+
     def switch_costume(self, name):
+        if name == "next backdrop":
```

Comment:
I will fix this in the merged code, but just for reference, these have to be turned into own methods, the value can not be hardcoded here, as this would break with translations. Note that if you switch Scratch to German, the value becomes "nächstes Bühnenbild".